psycopg/psycopg/types/enum.py

```diff
@@ -21,6 +24,43 @@
 EnumMapping: TypeAlias = Union[Mapping[E, str], Sequence[Tuple[E, str]], None]
 
 
+class EnumInfo(TypeInfo):
+    """Manage information about an enum type."""
+
+    def __init__(
+        self,
+        name: str,
+        oid: int,
+        array_oid: int,
+        labels: Sequence[str],
+    ):
+        super().__init__(name, oid, array_oid)
+        self.labels = labels
+        # Will be set by register_enum()
+        self.enum: Optional[Type[Enum]] = None
+
+    @classmethod
+    def _get_info_query(cls, conn: "BaseConnection[Any]") -> Query:
+        from ..sql import SQL
```

Comment:
maybe you can remove the lazy load here too?

Reply:
Most likely. Will tweak.